File: custom_waveform_doc/sin_wave.c
#include <math.h>
#include <stdio.h>

const double pi = acos(-1);

const double duration = 10;  // ms
const double amplitude_max = 100;  // uA
const char output_file[] = "sine.cwave";

int main() {
    int clock_div = (int)(ceil(duration * 1000 / 17.152 / 0x7fff));
    int num_of_samples = (int)(round(duration * 1000 / 17.152 / clock_div));

    FILE* fp = fopen(output_file, "w");
    for (int i = 0; i < num_of_samples; ++i)
        fprintf(fp, "%f\n", (1 - cos(i * pi / num_of_samples * 2)) * amplitude_max / 2);
    fclose(fp);

    printf("Generated %d samples. Please use %.3f us as sample interval.\n",
        num_of_samples, clock_div * 17.152);

    return 0;
}

File: custom_waveform_doc/sin_wave.py
#!/usr/bin/env python

import math


def main():
    duration = 10  # ms
    amplitude_max = 100  # uA
    output_file = 'sine.cwave'

    clock_div = int(math.ceil(duration * 1000 / 17.152 / 0x7fff))
    num_of_samples = int(round(duration * 1000 / 17.152 / clock_div))

    with open(output_file, 'w') as fp:
        for i in range(num_of_samples):
            print((1 - math.cos(i / num_of_samples * 2 * math.pi)) * amplitude_max / 2, file=fp)

    print('Generated %d samples. Please use %.3f us as sample interval.'
          % (num_of_samples, clock_div * 17.152))


if __name__ == '__main__':
    main()
